### PR TITLE
Improve input validation on Enterprise fees page

### LINKED ISSUE #9860: [Admin][Enterprise Fees] Improve validation of input
## Description

These are little leftovers from #9791 and exactly the same as #9836 for shipping methods. Maybe the code can be referenced instead of copying it?

On page `/admin/enterprise_fees` there is still room for improvement.

- For fee calculator 'Flexible rate' the 'max items' value is not validated as other fields, but it is set to '0' for invalid input while for 'Price sack' the amount values are validated as other fields and the error message is displayed for invalid input. Flexible rate should do the same.
- After clicking 'update' or 'create' invalid input field(s) should be highlighted (red border of input field) and all other updated information should be kept (including changes on other existing enterprise fees). Currently changes are lost and have to be entered again.

## Expected Behavior

- All input data should be validated.
- There should be an error message when invalid data is submitted.
- Invalid data should be highlighted.
- Updated information should be kept when submission of the form fails.

## Actual Behaviour

- Not all input data is validated.
- There is no error message when invalid data is submitted in the field 'max items' of the 'Flexible rate' calculator.
- Invalid data is not highlighted.
- Updated information is lost when submission of the form fails.

## Steps to Reproduce

Creating or changing an enterprise fee with 'Flexible rate':
1. As enterprise user or super admin user go to `/admin/enterprise_fees`
2. Select fee type, enter a name and select a tax category
3. Select the 'Flexible rate' calculator and click 'update'
4. Type invalid information into the 'max unit' field (e.g. the text 'invalid')
5. Click 'update'
6. See that the invalid value is set to 0

Creating or changing any shipping method:
1. As enterprise user or super admin user go to `/admin/enterprise fees`
2. Fill in required field, select any calculator and click 'update'
3. Update some fields (e.g. description) - do not click update yet
4. Type invalid information into any calculator field (e.g. the text 'invalid' in the 'amount' field)
5. Click 'update'
6. See that the updated information is lost
8. See that is no highlighting of the invalid input

## Animated Gif/Screenshot

Peek 2022-10-24 22-02

Peek 2022-10-24 22-01

## Severity
bug-s4: it's annoying, but you can use it

## Your Environment

* Version used:
* Browser name and version:
* Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by brolinr**
#### Improved enterprise fee validation of input? ''?

- Closes #9860 

#### Steps to reproduce as an admin?

- edit an enterprise fee
- choose flexible rate calculator
- click update
- type in an invalid value on 'max items' (e.g. some text or numbers with comma as separator)
- click update --> invalid input is highlighted

#### Release notes

Changelog Category: User facing changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: app/controllers/admin/enterprise_fees_controller.rb [hunk 1]
```diff
@@ -106,7 +106,7 @@ def check_enterprise_fee_input
           :preferred_flat_percent, :preferred_amount,
           :preferred_first_item, :preferred_additional_item,
           :preferred_minimal_amount, :preferred_normal_amount,
-          :preferred_discount_amount, :preferred_per_unit
+          :preferred_discount_amount, :preferred_per_unit, :preferred_max_items, :preffered_currency
```
Comment (jibees) [inline app/controllers/admin/enterprise_fees_controller.rb:109]: `preffered_currency `

I think there is a typo on `preffered`

**Review by jibees — CHANGES_REQUESTED** [review]
Thanks for this one. Small comments:
 - Title of the PR should be edited: we use the title as our Release notes, so they should be clear, concise and reflect what are the changes for the users
 - Commit d867ffe38c57241701b6acce86763d55e2158c81 should be deleted
 - Commit 340b41bbb1d164146073e9bde58ecfdc53d9fa62 should be squashed with the previous one (it's probably better for the git history to avoid this kind of commits which doesn't bring changes)

File: app/controllers/admin/enterprise_fees_controller.rb (hunk 1 above)
Comment (brolinr) [reply app/controllers/admin/enterprise_fees_controller.rb:109]: Thanks for the review and the comments.  I'll work on it 👍

**Comment by brolinr** [comment]
Hi @RachL . I am well thank you. I haven't yet found the time to apply the changes. I have been busy stydying for exams but I'd really appreciate it if you can help applying the comments from the review.

**Comment by RachL** [comment]
Hi @brolinr thank you for your feedback and don't worry: life happens :) Our team is currently quite busy, so I will close the PR to make the issue available for new contributors. Hopefully one of them will be able to pick-up your PR and apply the comments.

Thank you so much for your work until here!

**Pull request closed without merging**